src/Tests/dotnet-test.Tests/GivenDotnetTestBuildsAndRunsArtifactPostProcessing.cs

```diff
@@ -0,0 +1,249 @@
+                                    "--collect:SampleDataCollector",
+                                    $"--test-adapter-path:{(merge ? Path.GetDirectoryName(s_dataCollectorDll) : Path.GetDirectoryName(s_dataCollectorNoMergeDll))}",
+                                    $"--settings:{runsettings}",
+                                    "--diag:" + testInstance.Path + "/logs/");
+
+            result.ExitCode.Should().Be(0);
+            AssertOutput(result.StdOut, merge);
+        }
+
+        [Theory]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void ArtifactPostProcessing_VSTest_TestContainers(bool merge)
+        {
+            TestAsset testInstance = _testAssetsManager.CopyTestAsset("VSTestMultiProjectSolution", Guid.NewGuid().ToString())
+                .WithSource();
+
+            string runsettings = GetRunsetting(testInstance.Path);
+
+            new PublishCommand(Log, Path.Combine(testInstance.Path, "sln.sln")).Execute("/p:Configuration=Release").Should().Pass();
+
+            CommandResult result = new DotnetVSTestCommand(Log)
+              .WithWorkingDirectory(testInstance.Path)
+                                    .WithEnvironmentVariable(FeatureFlag.ARTIFACTS_POSTPROCESSING, "1")
+                                    .Execute(
+                                    Directory.GetFiles(testInstance.Path, "test1.dll", SearchOption.AllDirectories).SingleOrDefault(x => x.Contains("publish")),
+                                    Directory.GetFiles(testInstance.Path, "test2.dll", SearchOption.AllDirectories).SingleOrDefault(x => x.Contains("publish")),
+                                    Directory.GetFiles(testInstance.Path, "test3.dll", SearchOption.AllDirectories).SingleOrDefault(x => x.Contains("publish")),
+                                    "--collect:SampleDataCollector",
+                                    $"--testAdapterPath:{(merge ? Path.GetDirectoryName(s_dataCollectorDll) : Path.GetDirectoryName(s_dataCollectorNoMergeDll))}",
+                                    $"--settings:{runsettings}",
+                                    "--diag:" + testInstance.Path + "/logs/");
+
+            result.ExitCode.Should().Be(0);
+            AssertOutput(result.StdOut, merge);
+        }
+
+        private static void AssertOutput(string stdOut, bool merge)
+        {
+            List<string> output = new();
+            using StringReader reader = new(stdOut);
+            while (true)
+            {
+                string line = reader.ReadLine()?.Trim();
+                if (line is null) break;
+                output.Add(line);
+            }
+
+            if (merge)
+            {
+                Assert.Equal(string.Empty, output[output.Count - 3].Trim());
+                Assert.Equal("Attachments:", output[output.Count - 2].Trim());
+                string mergedFile = output[output.Count - 1].Trim();
+
+                var fileContent = new List<string>();
```

Comment:
Please use a `HashSet<string>` instead of the `List`, this way you don't need the `Distinct()`.

Reply:
I'll do